src/main/scala/exercises/05_Patterns.scala

```diff
@@ -56,15 +62,26 @@ object PatternMatching {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case str : String      => s"A string with length ${str.length}"
+      case i : Int if i > 0                   => "A positive integer"
+      case p : Person                         => s"A person with name: ${p.name}"
+      case seq : Seq[_] if seq.length > 10    => "Seq with more than 10 elements"
+      case v1::v2::tail                       => s"first: ${v1}, second: ${v2}, rest: ${tail}"
```

Comment:
I used another if statement here, but this is way more idiomatic. I wasn't able to find this Scala idiom. Good job!